dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java

```diff
@@ -59,157 +51,167 @@
+	
+	/** default asset store implementation */
+	private static final String DEFAULT_STORE_PREFIX = "ds:";
+	private static final String DEFAULT_STORE_IMPL = "org.dspace.storage.bitstore.impl.DSBitStore";
 
     /* Read in the asset stores from the config. */
     static
     {
-        List<Object> stores = new ArrayList<Object>();
-
-		// 'assetstore.dir' is always store number 0
-		String sAssetstoreDir = ConfigurationManager
-				.getProperty("assetstore.dir");
- 
-		// see if conventional assetstore or srb
-		if (sAssetstoreDir != null) {
-			stores.add(sAssetstoreDir); // conventional (non-srb)
-		} else if (ConfigurationManager.getProperty("srb.host") != null) {
-			stores.add(new SRBAccount( // srb
-					ConfigurationManager.getProperty("srb.host"),
-					ConfigurationManager.getIntProperty("srb.port"),
-					ConfigurationManager.getProperty("srb.username"),
-					ConfigurationManager.getProperty("srb.password"),
-					ConfigurationManager.getProperty("srb.homedirectory"),
-					ConfigurationManager.getProperty("srb.mdasdomainname"),
-					ConfigurationManager
-							.getProperty("srb.defaultstorageresource"),
-					ConfigurationManager.getProperty("srb.mcatzone")));
-		} else {
-			log.error("No default assetstore");
-		}
+    	ArrayList list = new ArrayList();
 
-		// read in assetstores .1, .2, ....
-		for (int i = 1;; i++) { // i == 0 is default above
-			sAssetstoreDir = ConfigurationManager.getProperty("assetstore.dir."
-					+ i);
-
-			// see if 'i' conventional assetstore or srb
-			if (sAssetstoreDir != null) { 		// conventional (non-srb)
-				stores.add(sAssetstoreDir);
-			} else if (ConfigurationManager.getProperty("srb.host." + i)
-					!= null) { // srb
-				stores.add(new SRBAccount(
-						ConfigurationManager.getProperty("srb.host." + i),
-						ConfigurationManager.getIntProperty("srb.port." + i),
-						ConfigurationManager.getProperty("srb.username." + i),
-						ConfigurationManager.getProperty("srb.password." + i),
-						ConfigurationManager
-								.getProperty("srb.homedirectory." + i),
-						ConfigurationManager
-								.getProperty("srb.mdasdomainname." + i),
-						ConfigurationManager
-								.getProperty("srb.defaultstorageresource." + i),
-						ConfigurationManager.getProperty("srb.mcatzone." + i)));
-			} else {
-				break; // must be at the end of the assetstores
-			}
-		}
-
-		// convert list to array
-		// the elements (objects) in the list are class
-		//   (1) String - conventional non-srb assetstore
-		//   (2) SRBAccount - srb assetstore
-		assetStores = new GeneralFile[stores.size()];
-		for (int i = 0; i < stores.size(); i++) {
-			Object o = stores.get(i);
-			if (o == null) { // I don't know if this can occur
-				log.error("Problem with assetstore " + i);
+        // Load the backwards compatible assetstore.dir's
+		list = initLegacyAssetstore(list);
+		
+		// if not already configured, configure asset stores
+		for (int j = 0; j < 100; j++)
```

Comment:
Setting a loop to 100 might lead to potential issues. Although I doubt that anybody would want to configure more then 100 configs it could be an issue. I would recommend creating a while loop and keep going until "ConfigurationManager.getProperty("assetstore." + j);" returns null.